### PR TITLE
tracking of written strings

### PR DESCRIPTION
**Pull request by Kumjami**
In order to be able to understand how users search for results we are going to track input strings. 
So everytime a user stop writting during 3 secs a new event is generated.

### CODE REVIEW

File: lib/analytics-helper/index.js [hunk 1]
```diff
@@ -96,6 +96,8 @@ export const addAnalyticsImpression = (item, dataLayer, impressionsTimestamp) =>
  * @returns {{event: string, preference_input: *, added_tags: *, displayed_tags: string}}
  */
 export const analyticsAddTagObject = (displayName, currentTags, context) => {
+  const initialTagIndex = currentTags.indexOf('Top inspiration');
```
Comment (tomgco) [inline lib/analytics-helper/index.js]: We should get the Tag `Top Inspiration` from the constant https://github.com/numo-labs/isearch-ui/blob/master/src/constants/default-tag.js

File: lib/search-bar/index.js [hunk 2]
```diff
@@ -30,11 +32,19 @@ export default class SearchBarContainer extends Component {
...
   handleInputChange (e) {
+    const input = e.target.value;
+    clearTimeout(keyPressTimeout);
+    keyPressTimeout = setTimeout(() => {
+      const analyticsObject = {
+        event: 'searchTyping',
+        searchInput: input
+      };
+      dataLayer.push(analyticsObject);
+    }, 3000);
```
Comment (tomgco) [inline lib/search-bar/index.js]: After every keypress, 3 seconds later we push an event? Maybe we could debounce on keypress and then submit the event after x amount of time?

File: lib/analytics-helper/index.js (hunk 1 above)
Comment (lennym) [inline lib/analytics-helper/index.js]: Beaten to the punch!

File: lib/search-bar/index.js (hunk 2 above)
Comment (Kumjami) [inline lib/search-bar/index.js]: It only triggers after 3 secs if there is no another keypressed within this time.

File: lib/analytics-helper/index.js (hunk 1 above)
Comment (Kumjami) [inline lib/analytics-helper/index.js]: oh!, will change it, thanks

File: lib/search-bar/index.js (hunk 2 above)
Comment (tomgco) [inline lib/search-bar/index.js]: Missed that, cool :+1:

File: lib/analytics-helper/index.js [hunk 3]
```diff
@@ -96,6 +96,8 @@ export const addAnalyticsImpression = (item, dataLayer, impressionsTimestamp) =>
...
+  if (initialTagIndex > -1) currentTags.splice(initialTagIndex, 1);
```
Comment (Kumjami) [inline lib/analytics-helper/index.js]: yes, @tomgco already pointed on that direction

**Comment by Kumjami** [comment]
@tomgco @lennym default tag readed from file

**Comment by tomgco** [comment]
LGTM!

**Pull request merged**